includes/Plugin_Context.php

```diff
@@ -85,4 +85,15 @@ public function location() {
 
 		return $path;
 	}
+
+	/**
+	 * Returns the plugin absolute path of the main file.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string The absolute path to the plugin main file.
+	 */
+	public function abspath() {
+		return $this->main_file;
+	}
```

Comment:
See my comment in #189, this shouldn't be needed. You can use `WP_PLUGIN_DIR` plus the `basename()` method.